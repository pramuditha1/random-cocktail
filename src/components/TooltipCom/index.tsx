import * as React from "react";
import Tooltip from "@mui/material/Tooltip";
import Lable from "../Lable";
export interface ToolTipPropTypes {
  text: string;
  maxLimit?: number;
}

const TooltipCom: React.FC<ToolTipPropTypes> = ({ text, maxLimit = 60 }) => {
  const isLongText = text.length > maxLimit;
  const displayText = isLongText ? `${text.slice(0, maxLimit)}...` : text;

  return (
    <Tooltip title={text} disableHoverListener={!isLongText}>
      <div>
        <Lable text={displayText} color="textSecondary" variant="caption" />
      </div>
    </Tooltip>
  );
};

export default TooltipCom;
