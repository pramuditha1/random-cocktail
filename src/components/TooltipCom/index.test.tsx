import React from "react";
import { render } from "@testing-library/react";
import TooltipCom, { ToolTipPropTypes } from "./index";

test("Tooltip component matches snapshot", () => {
  const mockProps: ToolTipPropTypes = {
    text: "Pour the corona into an 18oz beer glass pour the bacardi and lim...",
    maxLimit: 60
  };

  const { asFragment } = render(<TooltipCom {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
