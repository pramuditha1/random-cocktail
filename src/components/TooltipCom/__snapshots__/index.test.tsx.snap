// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Pour the corona into an 18oz beer glass pour the bacardi and lim..."
    class=""
    data-mui-internal-clone-element="true"
  >
    <div
      class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom css-10ylt8q-MuiTypography-root"
    >
      Pour the corona into an 18oz beer glass pour the bacardi and...
    </div>
  </div>
</DocumentFragment>
`;
